Open Judge System/Web/OJS.Web/Controllers/BaseController.cs

```diff
@@ -140,6 +139,14 @@ protected bool CheckIfUserHasContestCategoryPermissions(int categoryId)
                             x.Lecturers.Any(y => y.LecturerId == this.UserProfile.Id));
         }
 
+        protected bool CheckIfUserOwnsSubmission(int submissionId)
+        {
+            return this.User.IsAdmin() ||
```

Comment:
There isn't much point in checking if the user is an administrator here, because that check is already covered in CheckIfUserHasProblemPermissions